packages/orm/src/decorators/utils.ts

```diff
@@ -0,0 +1,37 @@
+import { Attribute } from '..';
+import { AttributeOptions, RelationalFieldOpts } from '../fields';
+import { AnyModel } from '../Model';
+import { ModelDescriptorsRegistry } from '../modelDescriptorsRegistry';
+import { Descriptors } from '../types';
+
+// Types of accepted descriptors
+type AttrsDescriptorFn = (opts: AttributeOptions) => Attribute;
+type BasicRelationalFieldDescriptorFn<DescriptorTypes extends Exclude<Descriptors, Attribute>> = (toModelName: string, relatedName?: string) => DescriptorTypes;
+type ComplexRelationalFieldDescriptorFn<DescriptorTypes extends Exclude<Descriptors, Attribute>> = (opts: RelationalFieldOpts) => DescriptorTypes;
+
+type TargetHandler = (target: AnyModel, propertyName: string) => void;
+
+// Types of possible decorator factories
+type AttrDecoratorFactory = (opts: AttributeOptions) => TargetHandler;
+type BasicRelationalFieldDecoratorFactory = (toModelName: string, relatedName?: string) => TargetHandler;
+type ComplexRelationalFieldDecoratorFactory = (opts: RelationalFieldOpts) => TargetHandler;
+
+export function registerDescriptor(descriptorFn: AttrsDescriptorFn): AttrDecoratorFactory;
```

Comment:
The HOF I will be probably exposing for creating new decorators on the consumers side. I have already overloaded type that accepts descriptor functions returning `Attribute` to this factory.